src/provider/model/Model.ts

```diff
@@ -175,13 +175,9 @@ export class Model {
         }, []);
     }
 
-    private onWindowSeen(identity: Identity): void {
+    private async onWindowCreated(identity: Identity): Promise<void> {
         this.getOrCreateExpectedWindow(identity);
```

Comment:
Can store the result and resuse, rather than calling the same function again a few lines down